pkg/vtpm/src/server.cpp

```diff
@@ -33,12 +41,160 @@ google::protobuf::uint32 readHdr(char *buf)
+"tpm2_startauthsession",
+"tpm2_policysecret",
+"tpm2_activatecredential",
+"tpm2_flushcontext",
+"tpm2_startauthsession",
+"tpm2_policysecret",
+"tpm2_import",
+"tpm2_flushcontext",
+"tpm2_load",
+"tpm2_evictcontrol",
+"tpm2_hmac",
+"tpm2_hash",
+"tpm2_sign",
+"tpm2_verifysignature",
+};
+
+static int
+sendResponse (int sock, eve_tools::EveTPMResponse &response)
+{
+    google::protobuf::uint32 size = response.ByteSize() + CODED_STRM_HDR_LEN;
+    char *resp_buffer = new char [size];
+    google::protobuf::io::ArrayOutputStream aos(resp_buffer, size);
+    CodedOutputStream *coded_output = new CodedOutputStream(&aos);
+    coded_output->WriteVarint32(response.ByteSize());
+    response.SerializeToCodedStream(coded_output);
+    int sent_bytes = send(sock, (void *)resp_buffer, size, 0);
+    delete(resp_buffer);
+    if (sent_bytes == size) {
+      return 0;
+    } else {
+      return -1;
+    }
+}
+
+static inline bool
+isCommandAllowed(string command_with_args)
+{
+  istringstream ss(command_with_args);
+  string command;
+  ss >> command;
+  auto it = find(allowed_commands.begin(), allowed_commands.end(), command);
+  if (it != allowed_commands.end()) {
+     return true;
+  }
+  return false;
+}
+
+static inline bool
+isFileNameAPath(string filename)
+{
+    return (filename.find('/') != filename.npos);
+}
+
+static int
+sanitizeCmdRequest(int sock, eve_tools::EveTPMRequest &request,
+                   eve_tools::EveTPMResponse &response)
+{
+    if (!isCommandAllowed(request.command())) {
+        cout << "Not a legal command, bailing out" << std::endl;
+        response.set_response("Command forbidden!");
+        sendResponse(sock, response);
+        return -1;
+    }
+    for (int i=0; i < request.expectedfiles_size(); i++) {
+        std::string expectedFile = request.expectedfiles(i);
+        if (isFileNameAPath(expectedFile)) {
+        response.set_response("output filename should not be a path!");
+        sendResponse(sock, response);
+        return -1;
+        }
+    }
+    for (int i=0; i < request.inputfiles_size(); i++) {
+        const eve_tools::File& file = request.inputfiles(i);
+        if (isFileNameAPath(file.name())) {
+            response.set_response("input filename should not be a path!");
+            sendResponse(sock, response);
+            return -1;
+        }
+    }
+    return 0;
+}
+
+//TBD: Change client library to send args in a list, rather in a single string
+//till then, split the given string into command and args. Once split, launch
+//the command using execve() inside a child. Parent waits for the client(i.e. command)
+//to finish.
+static int
+execCmd (string cmd) {
+    int i = 0;
+    string command_alone;
+    istringstream full_cmd(cmd);
+    const char *path[MAX_ARGS];
+    string args[MAX_ARGS];
+
+    full_cmd >> command_alone;
+    string command_with_path = BIN_PATH + command_alone;
+
+    path[i++] = command_with_path.c_str();
+    while (full_cmd >> args[i] && i < MAX_ARGS) {
```

Comment:
How about returning an error if more than MAX_ARGS?

Reply:
fixed it in the latest diff.